### PR TITLE
Update test router

### PR DESCRIPTION
**Pull request by arajguha**
## Description of the change
- added wrapper function for testRouter
- added test cases
- added check on content-type from destination response
> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Comment by arajguha** [comment]
`network-layer-updates` is merged into this branch.

File: __tests__/testRouter.test.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+          }
+        }
+      };
+      it(`Input: ${index} should return proper response`, async () => {
+        try {
+          await handleTestEvent(ctxMock, dest.toLowerCase());
+          expect(ctxMock.body).toEqual(outputData[index]);
+        }
+        catch (err) {
+          // should not enter this block
```
Comment (sanpj2292) [inline __tests__/testRouter.test.js:40]: Say there is an exception with a case or due to a change, I think the test-case will get passed because we've caught the error already. Can you verify if that is the case ?

File: routerUtils.js [hunk 2]
```diff
@@ -27,6 +27,7 @@ async function sendToDestination(destination, payload) {
   if (resp.success) {
     const { response } = resp;
     parsedResponse = {
+      headers: response.headers,
```
Comment (sanpj2292) [inline routerUtils.js:30]: why is this added ?

Comment (arajguha) [reply routerUtils.js:30]: for checking the response type. For eg: if the content-type is image/gif, we are setting the response to an empty string.

Comment (sanpj2292) [reply routerUtils.js:30]: Got it 👍🏼

File: __tests__/testRouter.test.js (hunk 1 above)
Comment (arajguha) [reply __tests__/testRouter.test.js:40]: yes actually. should not be wrapping test function call in try-catch block. will update this part.

File: testRouter.js [hunk 3]
```diff
@@ -44,149 +40,167 @@ const transformDestination = dest => {
...
+              const parsedResponse = await sendToDestination(dest, payload);
+
+              let contentType = "";
+              let response = "";
+              if (parsedResponse.headers) {
+                contentType = parsedResponse.headers["content-type"];
+                if (
+                  contentType.includes("text") ||
+                  contentType.toLowerCase().includes("application/xml") ||
+                  contentType.toLowerCase().includes("application/json")
```
Comment (saikumarrs) [inline testRouter.js:146]: I would create a separate list of supported content types.

Comment (arajguha) [reply testRouter.js:146]: makes sense. will put them in a list

**Review by saikumarrs — DISMISSED** [review]

**Review by saikumarrs — APPROVED** [review]
